packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -636,10 +634,16 @@ - (FLTPositionMessage *)position:(FLTTextureMessage *)input error:(FlutterError
   return result;
 }
 
-- (void)seekTo:(FLTPositionMessage *)input error:(FlutterError **)error {
+- (void)seekTo:(FLTPositionMessage *)input
+    completion:(void (^)(FlutterError *_Nullable))completion {
   FLTVideoPlayer *player = self.playersByTextureId[input.textureId];
-  [player seekTo:input.position.intValue];
-  [self.registry textureFrameAvailable:input.textureId.intValue];
+  [player seekTo:input.position.intValue
+      completionHandler:^(BOOL finished) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+          [self.registry textureFrameAvailable:input.textureId.intValue];
+          completion(nil);
```

Comment:
may wanna pass `finished` to dart side?

Reply:
It is possible, however that would require an interface change, right?

Reply:
isn't changing from sync to async already an interface change? I think the question here is, if it's not `finished`, what do we expect UI to behave?

Reply:
It is not an interface change because the method channels calls are always async by its nature, right?

According to the documentation of AVPlayer, finished as true means that the seekTo completed properly, while finished as false means that seekTo was aborted because of another seekTo request. So in both cases it means the seekTo is not happening anymore. I agree it can be an useful information though, but I tried to go with a slightly simpler improvement as my first contribution :D

I can implement the interface change if needed, of course. Let me know.

Reply:
gotcha. when seekTo fails, will the UI revert back to the original position before the move?

Reply:
In theory we could pass it to the Dart side (changing the Pigeon interface definition) and then drop it [here](https://github.com/flutter/plugins/blob/8eede17eea3aacde4c2c8e89bd721ea06e9acb02/packages/video_player/video_player_avfoundation/lib/src/avfoundation_video_player.dart#L107), which would still be an implementation-package-local change. I don't think there's much value in doing that since we have no plans to use that information, but @hellohuanlin if you feel strongly about adding it, it would be easy enough to do.

This PR definitely should not become a breaking change to the public seek API.

Reply:
gotcha. i don't have an strong opinion. I just wanted to make sure the UI can revert back to the original position if `finished` is false. Sounds like we don't need this flag to achieve this.

Reply:
In general, the position handling in `video_player` is not great right now; it can get out of sync with native for short periods quite easily, and is primary polling based. It needs an overhaul, and there are issues tracking that, but it's way out of scope here.

Reply:
In this case, if I may ask, what are the following steps I need to take now?
I see that "submit-queue" is still failling, but I could not find out what that means.

Reply:
`submit-queue` means the tree is closed, and isn't related to the PR; you don't need to worry about it.